audio_api/services/tts_service.py

```
@@ -1,95 +1,228 @@
+        )
+
+        @retry_decorator
+        async def _generate_with_retry():
+            try:
+                return await self._generate_gemini_audio(request)
+            except Exception as e:
+                error_msg = str(e).lower()
+
+                # Categorize errors for appropriate retry behavior
+                if "invalid_argument" in error_msg or "validation" in error_msg:
+                    raise TTSValidationError(f"Invalid request parameters: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)